ls1intum/Artemis
# [Bugfix] ComponentCanDeactivate Trigger

### Checklist

### Motivation and Context
1. Clone dialog should not activate the "ComponentCanDeactivate" dialog, see #1966
2. The "ComponentCanDeactivate" dialogue was also triggered if the student is not registered to an exam, or if the student was trying to submit early.

### Description
1. Do not trigger the dialogue, if the student is handing in early or if there is no student exam
2. Fix #1966 using <iframe>

### Steps for Testing
1. Log in to Artemis
2. Go to an exam where you are not registered. 
3. Navigate away from the exam. 
=> The ComponentCanDeactivate should not trigger
4. Now go to an exam where you are registered. 
5. Try cloning a programing exercise using clone in sourcetree
=> ComponentCanDeactivate should not trigger
6. Hand in early
=> ComponentCanDeactivate should not trigger
